### ISSUE TITLE
Results Caching

### ISSUE THREAD

**Issue by AdamBelfki3** [ndif-team/workbench#38]
This ticket aims to Tool results redering more efficient for the user by sending fewer requests by the api to NDIF. This would involve capturing all the logits, at every layer and token position, and compute all the metrics together and store them in memory on the api side. This way, if the user switches between metrics visualized, the api is able to provide it instantly without relying on NDIF. The cache per prompt would be cleared after some time interval.

**Comment by MichaelRipa** [comment]
Another neat feature related to this would be having a way of setting a cached prompt that you can execute from. I modified the html to create a hacky POC of what I mean:

<img width="1136" height="756" alt="Image" src="https://github.com/user-attachments/assets/ce0c945c-5635-4982-b98f-d8a3d16ed4f7" />

The first prompt would be cached, allowing you to reuse its intermediate values, which could speed up the remote requests and improve user experience. Also I imagine it would be nice to have a way of toggling between hiding and displaying the heatmap of the first prompt.
